main.tf

```diff
@@ -56,6 +56,17 @@ resource "azurerm_container_app_environment_dapr_component" "dapr" {
   }
 }
 
+resource "azurerm_container_app_environment_storage" "storage" {
+  for_each = { for storage in var.env_storage : storage.name => storage }
```

Comment:
Since `var.env_storage`'s type is `map(object({...}))`, the map's key should be static string and object's `name` property could contain `known after apply` value, we should use `var.env_storage` directly here.